service/history/handler.go

```diff
@@ -1984,8 +1943,11 @@ func (h *Handler) PollWorkflowExecutionUpdate(
 
 func (h *Handler) StreamWorkflowReplicationMessages(
 	server historyservice.HistoryService_StreamWorkflowReplicationMessagesServer,
-) (retError error) {
-	defer metrics.CapturePanic(h.logger, h.metricsHandler, &retError)
+) (retErr error) {
+	defer metrics.CapturePanic(h.logger, h.metricsHandler, &retErr)
+
+	// NOTE we need to handle this ourselves here since this is a stream endpoint
+	// and the `lifecycleInterceptor` does not apply here
 	h.startWG.Wait()
```

Comment:
This is a very good catch, but I'm worried about others missing it with new streaming methods. What about using a [StreamServerInterceptor](https://github.com/grpc/grpc-go/blob/master/examples/features/interceptor/README.md#stream-interceptor-1)?

Reply:
I briefly tried to write a StreamServerInterceptor but couldn't figure out how to obtain a reference to the `Handler`. I could give it another look.